Error after moving all services to unRaid.

They were all working when they were stand alone dockers on a ubuntu box. I moved them all unRaid yesterday and noticed this error. If I comment out Radarr from the config. Only the top error goes away. Note - Notifiarr is in a docker on the ubuntu box.
```
[ERROR] 2021/11/16 18:14:15 Radarr URL fixed: http://192.168.1.241:7878/ -> http://192.168.1.241:7878 (continuing)
[ERROR] 2021/11/16 18:12:40 Go Panic! This is a bug please report it on github: https://github.com/Notifiarr/notifiarr/issues/new

runtime error: invalid memory address or nil pointer dereference

goroutine 1 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
main.main.func1()
	/notifiarr/main.go:21 +0x3d
panic({0x87eda0, 0xc9d800})
	/usr/local/go/src/runtime/panic.go:1038 +0x215
github.com/Notifiarr/notifiarr/pkg/apps.(*Apps).Setup(0xc000240000, 0x8cd460)
	/notifiarr/pkg/apps/apps.go:177 +0x374
github.com/Notifiarr/notifiarr/pkg/configfile.(*Config).Get(0xc0002304e0, {0x90c498, 0x16}, {0x9030c7, 0x2})
	/notifiarr/pkg/configfile/config.go:95 +0x1bb
github.com/Notifiarr/notifiarr/pkg/client.(*Client).loadConfiguration(0xc000214980)
	/notifiarr/pkg/client/start.go:137 +0x1dd
github.com/Notifiarr/notifiarr/pkg/client.(*Client).start(0xc000214980)
	/notifiarr/pkg/client/start.go:81 +0x36
github.com/Notifiarr/notifiarr/pkg/client.Start()
	/notifiarr/pkg/client/start.go:76 +0xae
main.main()
```

--- a/pkg/apps/apps.go
+++ b/pkg/apps/apps.go
@@ -159,7 +159,7 @@ func (a *Apps) InitHandlers() {
 
 // Setup creates request interfaces and sets the timeout for each server.
 // This is part of the config/startup init.
-func (a *Apps) Setup(timeout time.Duration) error { //nolint:funlen,gocognit
+func (a *Apps) Setup(timeout time.Duration) error {
 	if a.DebugLog == nil {
 		a.DebugLog = log.New(io.Discard, "", 0)
 	}
@@ -168,70 +168,32 @@ func (a *Apps) Setup(timeout time.Duration) error { //nolint:funlen,gocognit
 		a.ErrorLog = log.New(io.Discard, "", 0)
 	}
 
-	for i := range a.Lidarr {
-		if a.Lidarr[i].Config == nil || a.Lidarr[i].Config.URL == "" {
-			return fmt.Errorf("%w: missing url: Lidarr config %d", ErrInvalidApp, i+1)
-		}
-
-		a.Lidarr[i].Debugf = a.DebugLog.Printf
-		a.Lidarr[i].Errorf = a.ErrorLog.Printf
-		a.Lidarr[i].setup(timeout)
+	if err := a.setupLidarr(timeout); err != nil {
+		return err
 	}
 
-	for i := range a.Radarr {
-		if a.Radarr[i].Config == nil || a.Radarr[i].Config.URL == "" {
-			return fmt.Errorf("%w: missing url: Radarr config %d", ErrInvalidApp, i+1)
-		}
-
-		a.Radarr[i].Debugf = a.DebugLog.Printf
-		a.Radarr[i].Errorf = a.ErrorLog.Printf
-		a.Radarr[i].setup(timeout)
+	if err := a.setupRadarr(timeout); err != nil {
+		return err
 	}
 
-	for i := range a.Readarr {
-		if a.Readarr[i].Config == nil || a.Readarr[i].Config.URL == "" {
-			return fmt.Errorf("%w: missing url: Readarr config %d", ErrInvalidApp, i+1)
-		}
-
-		a.Readarr[i].Debugf = a.DebugLog.Printf
-		a.Readarr[i].Errorf = a.ErrorLog.Printf
-		a.Readarr[i].setup(timeout)
+	if err := a.setupReadarr(timeout); err != nil {
+		return err
 	}
 
-	for i := range a.Sonarr {
-		if a.Sonarr[i].Config == nil || a.Sonarr[i].Config.URL == "" {
-			return fmt.Errorf("%w: missing url: Sonarr config %d", ErrInvalidApp, i+1)
-		}
+	if err := a.setupSonarr(timeout); err != nil {
+		return err
+	}
 
-		a.Sonarr[i].Debugf = a.DebugLog.Printf
-		a.Sonarr[i].Errorf = a.ErrorLog.Printf
-		a.Sonarr[i].setup(timeout)
+	if err := a.setupDeluge(timeout); err != nil {
+		return err
+	}
+
+	if err := a.setupQbit(timeout); err != nil {
+		return err
 	}
 
 	for i := range a.SabNZB {
 		a.SabNZB[i].setup(timeout)
-	}
-
-	for i := range a.Deluge {
-		if a.Deluge[i] == nil || a.Deluge[i].Config == nil || a.Deluge[i].Config.URL == "" {
-			return fmt.Errorf("%w: missing url: Deluge config %d", ErrInvalidApp, i+1)
-		}
-
-		// a.Deluge[i].Debugf = a.DebugLog.Printf
-		if err := a.Deluge[i].setup(timeout); err != nil {
-			return err
-		}
-	}
-
-	for i := range a.Qbit {
-		if a.Qbit[i].Config == nil || a.Qbit[i].URL == "" {
-			return fmt.Errorf("%w: missing url: Qbit config %d", ErrInvalidApp, i+1)
-		}
-
-		// a.Qbit[i].Debugf = a.DebugLog.Printf
-		if err := a.Qbit[i].setup(timeout); err != nil {
-			return err
-		}
 	}
 
 	a.Tautulli.setup(timeout)

--- a/pkg/apps/apps.go
+++ b/pkg/apps/apps.go
@@ -57,6 +57,7 @@ var (
 	ErrNonZeroID = fmt.Errorf("provided ID must be non-zero")
 	// ErrWrongCount is returned when an app returns the wrong item count.
 	ErrWrongCount = fmt.Errorf("wrong item count returned")
+	ErrInvalidApp = fmt.Errorf("invalid application configuration provided")
 )
 
 // APIHandler is our custom handler function for APIs.
@@ -158,7 +159,7 @@ func (a *Apps) InitHandlers() {
 
 // Setup creates request interfaces and sets the timeout for each server.
 // This is part of the config/startup init.
-func (a *Apps) Setup(timeout time.Duration) error {
+func (a *Apps) Setup(timeout time.Duration) error { //nolint:funlen,gocognit
 	if a.DebugLog == nil {
 		a.DebugLog = log.New(io.Discard, "", 0)
 	}
@@ -167,28 +168,44 @@ func (a *Apps) Setup(timeout time.Duration) error {
 		a.ErrorLog = log.New(io.Discard, "", 0)
 	}
 
+	for i := range a.Lidarr {
+		if a.Lidarr[i].Config == nil || a.Lidarr[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Lidarr config %d", ErrInvalidApp, i+1)
+		}
+
+		a.Lidarr[i].Debugf = a.DebugLog.Printf
+		a.Lidarr[i].Errorf = a.ErrorLog.Printf
+		a.Lidarr[i].setup(timeout)
+	}
+
 	for i := range a.Radarr {
+		if a.Radarr[i].Config == nil || a.Radarr[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Radarr config %d", ErrInvalidApp, i+1)
+		}
+
 		a.Radarr[i].Debugf = a.DebugLog.Printf
 		a.Radarr[i].Errorf = a.ErrorLog.Printf
 		a.Radarr[i].setup(timeout)
 	}
 
 	for i := range a.Readarr {
+		if a.Readarr[i].Config == nil || a.Readarr[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Readarr config %d", ErrInvalidApp, i+1)
+		}
+
 		a.Readarr[i].Debugf = a.DebugLog.Printf
 		a.Readarr[i].Errorf = a.ErrorLog.Printf
 		a.Readarr[i].setup(timeout)
 	}
 
 	for i := range a.Sonarr {
+		if a.Sonarr[i].Config == nil || a.Sonarr[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Sonarr config %d", ErrInvalidApp, i+1)
+		}
+
 		a.Sonarr[i].Debugf = a.DebugLog.Printf
 		a.Sonarr[i].Errorf = a.ErrorLog.Printf
 		a.Sonarr[i].setup(timeout)
-	}
-
-	for i := range a.Lidarr {
-		a.Lidarr[i].Debugf = a.DebugLog.Printf
-		a.Lidarr[i].Errorf = a.ErrorLog.Printf
-		a.Lidarr[i].setup(timeout)
 	}
 
 	for i := range a.SabNZB {
@@ -196,6 +213,10 @@ func (a *Apps) Setup(timeout time.Duration) error {
 	}
 
 	for i := range a.Deluge {
+		if a.Deluge[i] == nil || a.Deluge[i].Config == nil || a.Deluge[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Deluge config %d", ErrInvalidApp, i+1)
+		}
+
 		// a.Deluge[i].Debugf = a.DebugLog.Printf
 		if err := a.Deluge[i].setup(timeout); err != nil {
 			return err
@@ -203,6 +224,10 @@ func (a *Apps) Setup(timeout time.Duration) error {
 	}
 
 	for i := range a.Qbit {
+		if a.Qbit[i].Config == nil || a.Qbit[i].URL == "" {
+			return fmt.Errorf("%w: missing url: Qbit config %d", ErrInvalidApp, i+1)
+		}
+
 		// a.Qbit[i].Debugf = a.DebugLog.Printf
 		if err := a.Qbit[i].setup(timeout); err != nil {
 			return err

--- a/pkg/apps/downloaders.go
+++ b/pkg/apps/downloaders.go
@@ -33,6 +33,21 @@ type TautulliConfig struct {
 	APIKey   string        `toml:"api_key"`
 }
 
+func (a *Apps) setupDeluge(timeout time.Duration) error {
+	for i := range a.Deluge {
+		if a.Deluge[i] == nil || a.Deluge[i].Config == nil || a.Deluge[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Deluge config %d", ErrInvalidApp, i+1)
+		}
+
+		// a.Deluge[i].Debugf = a.DebugLog.Printf
+		if err := a.Deluge[i].setup(timeout); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
 func (d *DelugeConfig) setup(timeout time.Duration) (err error) {
 	d.Deluge, err = deluge.New(*d.Config)
 	if err != nil {
@@ -41,6 +56,21 @@ func (d *DelugeConfig) setup(timeout time.Duration) (err error) {
 
 	if d.Timeout.Duration == 0 {
 		d.Timeout.Duration = timeout
+	}
+
+	return nil
+}
+
+func (a *Apps) setupQbit(timeout time.Duration) error {
+	for i := range a.Qbit {
+		if a.Qbit[i].Config == nil || a.Qbit[i].URL == "" {
+			return fmt.Errorf("%w: missing url: Qbit config %d", ErrInvalidApp, i+1)
+		}
+
+		// a.Qbit[i].Debugf = a.DebugLog.Printf
+		if err := a.Qbit[i].setup(timeout); err != nil {
+			return err
+		}
 	}
 
 	return nil

--- a/pkg/apps/lidarr.go
+++ b/pkg/apps/lidarr.go
@@ -52,6 +52,20 @@ type LidarrConfig struct {
 	Errorf func(string, ...interface{}) `toml:"-" xml:"-"`
 }
 
+func (a *Apps) setupLidarr(timeout time.Duration) error {
+	for i := range a.Lidarr {
+		if a.Lidarr[i].Config == nil || a.Lidarr[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Lidarr config %d", ErrInvalidApp, i+1)
+		}
+
+		a.Lidarr[i].Debugf = a.DebugLog.Printf
+		a.Lidarr[i].Errorf = a.ErrorLog.Printf
+		a.Lidarr[i].setup(timeout)
+	}
+
+	return nil
+}
+
 func (r *LidarrConfig) setup(timeout time.Duration) {
 	r.Lidarr = lidarr.New(r.Config)
 	if r.Timeout.Duration == 0 {

--- a/pkg/apps/radarr.go
+++ b/pkg/apps/radarr.go
@@ -54,6 +54,20 @@ type RadarrConfig struct {
 	Errorf func(string, ...interface{}) `toml:"-" xml:"-"`
 }
 
+func (a *Apps) setupRadarr(timeout time.Duration) error {
+	for i := range a.Radarr {
+		if a.Radarr[i].Config == nil || a.Radarr[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Radarr config %d", ErrInvalidApp, i+1)
+		}
+
+		a.Radarr[i].Debugf = a.DebugLog.Printf
+		a.Radarr[i].Errorf = a.ErrorLog.Printf
+		a.Radarr[i].setup(timeout)
+	}
+
+	return nil
+}
+
 func (r *RadarrConfig) setup(timeout time.Duration) {
 	r.Radarr = radarr.New(r.Config)
 	if r.Timeout.Duration == 0 {

--- a/pkg/apps/readarr.go
+++ b/pkg/apps/readarr.go
@@ -47,6 +47,20 @@ type ReadarrConfig struct {
 	Errorf func(string, ...interface{}) `toml:"-" xml:"-"`
 }
 
+func (a *Apps) setupReadarr(timeout time.Duration) error {
+	for i := range a.Readarr {
+		if a.Readarr[i].Config == nil || a.Readarr[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Readarr config %d", ErrInvalidApp, i+1)
+		}
+
+		a.Readarr[i].Debugf = a.DebugLog.Printf
+		a.Readarr[i].Errorf = a.ErrorLog.Printf
+		a.Readarr[i].setup(timeout)
+	}
+
+	return nil
+}
+
 func (r *ReadarrConfig) setup(timeout time.Duration) {
 	r.Readarr = readarr.New(r.Config)
 	if r.Timeout.Duration == 0 {

--- a/pkg/apps/sonarr.go
+++ b/pkg/apps/sonarr.go
@@ -51,6 +51,20 @@ type SonarrConfig struct {
 	Errorf func(string, ...interface{}) `toml:"-" xml:"-"`
 }
 
+func (a *Apps) setupSonarr(timeout time.Duration) error {
+	for i := range a.Sonarr {
+		if a.Sonarr[i].Config == nil || a.Sonarr[i].Config.URL == "" {
+			return fmt.Errorf("%w: missing url: Sonarr config %d", ErrInvalidApp, i+1)
+		}
+
+		a.Sonarr[i].Debugf = a.DebugLog.Printf
+		a.Sonarr[i].Errorf = a.ErrorLog.Printf
+		a.Sonarr[i].setup(timeout)
+	}
+
+	return nil
+}
+
 func (r *SonarrConfig) setup(timeout time.Duration) {
 	r.Sonarr = sonarr.New(r.Config)
 	if r.Timeout.Duration == 0 {

